src/endpoints/Endpoint.js

```diff
@@ -86,37 +83,45 @@ class Endpoint {
     return `${this.stateName}Payload`
   }
 
-  fetchEndpointAction (urlParams = {}, options = {}) {
+  requestAction (urlParams = {}, options = {}) {
     return (dispatch, getState) => {
       if (getState()[this.name].isFetching) {
         return
       }
 
-      dispatch(this.requestAction())
-
-      let formattedURL = format(this.url, urlParams)
+      const formattedURL = format(this.url, urlParams)
       /*
        todo:  check if there are any paramters left in the url: formattedURL.match(/{(.*)?}/)
        currently this does not work as unused paramaters are just removed from the url
        */
 
+      const lastUrl = getState()[this.name].requestUrl
+
+      if (lastUrl && lastUrl === formattedURL) {
+        // Use "cached"
```

Comment:
I think you your comment wants to say, "Use the current payload which we already have in place in the redux store.". "Use cached" sounds for me like "We have stored sth somewhere and load that in now instead of fetching from server again."?